src/project-wide-dependency-checker.js

```diff
@@ -77,4 +94,76 @@ module.exports = class ProjectWideDependencyChecker {
 
     throw new SilentError(message);
   }
+
+  check(constraints) {
+    const names = Object.keys(constraints);
+    const addons = this.filterAddonsByNames(names);
+    const node_modules = Object.create(null);
+
+    for (let name in addons) {
+      const found = addons[name];
+      const versions = found.map(addon => addon.pkg.version);
+
+      const constraint = constraints[name];
+      const missing = versions.length === 0;
+      const isSatisfied =
+        !missing &&
+        versions.every(version => semver.satisfies(version, constraint));
+
+      let message;
+      if (isSatisfied) {
+        message = '';
+      } else if (missing) {
+        message = `${name} was not found, expected version: ${constraint}`;
+      } else {
+        message = `${name} expected version: ${constraint} but got version${
+          versions.length > 1 ? 's' : ''
+        }: ${versions.join(', ')}`;
+      }
+
+      node_modules[name] = {
```

Comment:
Does snake_case confuses linters?

Reply:
No doesn't confuse the current lit setup. This just felt right given the context.